homeassistant/components/sia/binary_sensor.py

```diff
@@ -98,66 +170,11 @@ def handle_last_state(self, last_state: State | None) -> None:
             elif last_state.state == STATE_UNAVAILABLE:
                 self._attr_available = False
 
-
-class SIABinarySensorMoisture(SIABinarySensorBase):
-    """Class for Moisture Binary Sensors."""
-
-    def __init__(
-        self,
-        entry: ConfigEntry,
-        account_data: dict[str, Any],
-        zone: int,
-    ) -> None:
-        """Initialize a Moisture binary sensor."""
-        super().__init__(entry, account_data, zone, BinarySensorDeviceClass.MOISTURE)
-        self._attr_entity_registry_enabled_default = False
-
-    def update_state(self, sia_event: SIAEvent) -> None:
-        """Update the state of the binary sensor."""
-        new_state = MOISTURE_CODE_CONSEQUENCES.get(sia_event.code, None)
-        if new_state is not None:
-            _LOGGER.debug("New state will be %s", new_state)
-            self._attr_is_on = new_state
-
-
-class SIABinarySensorSmoke(SIABinarySensorBase):
-    """Class for Smoke Binary Sensors."""
-
-    def __init__(
-        self,
-        entry: ConfigEntry,
-        account_data: dict[str, Any],
-        zone: int,
-    ) -> None:
-        """Initialize a Smoke binary sensor."""
-        super().__init__(entry, account_data, zone, BinarySensorDeviceClass.SMOKE)
-        self._attr_entity_registry_enabled_default = False
-
-    def update_state(self, sia_event: SIAEvent) -> None:
-        """Update the state of the binary sensor."""
-        new_state = SMOKE_CODE_CONSEQUENCES.get(sia_event.code, None)
-        if new_state is not None:
-            _LOGGER.debug("New state will be %s", new_state)
-            self._attr_is_on = new_state
-
-
-class SIABinarySensorPower(SIABinarySensorBase):
-    """Class for Power Binary Sensors."""
-
-    def __init__(
-        self,
-        entry: ConfigEntry,
-        account_data: dict[str, Any],
-    ) -> None:
-        """Initialize a Power binary sensor."""
-        super().__init__(
-            entry, account_data, SIA_HUB_ZONE, BinarySensorDeviceClass.POWER
-        )
-        self._attr_entity_registry_enabled_default = True
-
-    def update_state(self, sia_event: SIAEvent) -> None:
+    def update_state(self, sia_event: SIAEvent) -> bool:
         """Update the state of the binary sensor."""
-        new_state = POWER_CODE_CONSEQUENCES.get(sia_event.code, None)
-        if new_state is not None:
-            _LOGGER.debug("New state will be %s", new_state)
-            self._attr_is_on = new_state
+        new_state = self.entity_description.code_consequences.get(sia_event.code, None)
```

Comment:
`None` is the default value returned if the key is missing in the dict when using `dict.get`.
```suggestion
        new_state = self.entity_description.code_consequences.get(sia_event.code)
```